File: chaeri/괄호변환.py
#분리하기
def divide(w):
    start = 0
    end = 0
    for i in range(len(w)):
        if w[i] == '(':
            start += 1
        else:
            end += 1
        if start == end:
            return w[:i + 1], w[i + 1:]


    
def solution(p):
    answer = ''
    return answer

File: taejoon/week8/괄호 변환.js
function solution(p) {
  if (p.length === 0) return '';

  const a = p[0];
  const b = p[0] === '(' ? ')' : '(';

  let a_count = 1;
  let b_count = 0;
  let idx = 1;

  while (a_count !== b_count) {
    if (p[idx] === a) a_count++;
    if (p[idx] === b) b_count++;
    idx++;
  }

  let u = p.slice(0, idx);
  let w = p.slice(idx);

  var answer = '';
  return answer;
}

function isRightString(str) {
  let stack = [];
  let x = str.split('');
  stack.push[x.shift()];
  while (x.length > 0) {
    if (stack[stack.length - 1] === '(' && x[0] === ')') {
      stack.pop();
      x.shift();
    } else {
      stack.push(x.shift());
    }
  }
  return stack.lenght === 0;
}
